### ISSUE TITLE
Sequence Notes caron

### ISSUE THREAD

**Issue by JoeCohen** [MushroomObserver/mushroom-observer#1808]
Certain string values in Sequence Notes cause `Mysql2::Error: Incorrect string value`. See example below.
###Tasks
- [x] Find short string that provokes the error
- [x] Write test (Red)
- [x] Make test Green
- [x] Rubocop
- [x] PR

**Comment by JoeCohen** [comment]
Locally editing Sequence Notes to be:
```
blah blah

The subject’s ITS sequence is highly similar to JF834342 Russula aff. firmula, CA , which is in the Russula firmula species complex according to the phylogenetic analysis of Caboň et. al. 2017. Based on sequencing results and morphology the subject is in the Russula firmula group.
```
Threw this:
```ruby
ActiveRecord::StatementInvalid in SequencesController#update
Mysql2::Error: Incorrect string value: '\xC5\x88 et....' for column 'notes' at row 1
Rails.root: /Users/joe/mushroom-observer

[Application Trace](http://localhost:3000/298551?q=1oQiQ#) | [Framework Trace](http://localhost:3000/298551?q=1oQiQ#) | [Full Trace](http://localhost:3000/298551?q=1oQiQ#)
Request
Parameters:

{"utf8"=>"✓",
 "_method"=>"patch",
 "authenticity_token"=>"[FILTERED]",
 "sequence"=>
  {"locus"=>"ITS",
   "bases"=>
    "AGGATCATTATTGAATAAACTTGACGTGGTTGTAGCTGACTCTCTCGGGAGTATGTGCTCGCTCGTCATTTTTATCTTTCCACCTGTGCACTTTTTGTAGATTTAGATGTGACTTTCTGAGGCAACTCGGTTGGGAGGAATGTCAATTCGTTGGCTTTCCTTGTATGTCCAAATCTATGTTTTTATACACTCAAAAAAATGTAACAGAATGTATACTAGGCCTTGTGCCTATAAATTATATACAACTTTCAGCAACGGATCTCTTGGCTCTCGCATCGATGAAGAACGCAGCGAAATGCGATAAGTAATGTGAATTGCAGAATTCAGTGAATCATCGAATCTTTGAACGCACCTTGCGCCCCTTGGTATTCCGAGGGGCATGCCTGTTTGAGTGTCATTAAATTCTCAACCTTTACTAGCTTTTTGTGAGCTATGTAATGGCTTGGATGTGGGGGTCTTTTTGCTGGCTTCGAAAGAAATCGGCTCCCCTGAAATATATTAGCTGGTGCCCCCTGCGTGGACCGTCTACTGGTGTGATAATTATCTATGCCGTTAGATGTCTGCTTTAAATAATGGGGATGCGCTGCTTCTAATCGTCCACTTAGGACAATTATTGACTATTGACCTCAATCA",
   "archive"=>"",
   "accession"=>"",
   "notes"=>
    "blah blah\n" +
    "\n" +
    "The subject’s ITS sequence is highly similar to JF834342 Russula aff. firmula, CA , which is in the Russula firmula species complex according to the phylogenetic analysis of Caboň et. al. 2017. Based on sequencing results and morphology the subject is in the Russula firmula group."},
 "commit"=>"Update",
 "back"=>"298551",
 "q"=>"1oQiQ",
 "id"=>"158"}
```

**Comment by JoeCohen** [comment]
The `ň` alone  -- **pasted from the [Comment above](https://github.com/MushroomObserver/mushroom-observer/issues/1808#issuecomment-1892990543) or from this line**-- causes the error.
BUT it's character-set dependent. This `ñ` (which I entered from my keyboard) works fine.

**Comment by JoeCohen** [comment]
In dev mode, when trying to add a sequence to an Obs, with sequence notes ==  'ň', it blows up upon executing this SQL, which looks like good sql to me.
```sql
INSERT INTO `sequences` (`observation_id`, `user_id`, `locus`, `bases`, `archive`, `accession`, `notes`, `created_at`, `updated_at`) VALUES (516715, 4468, 'caron', 'atatat', '', '', 'ň', '2024-01-16 20:22:44', '2024-01-16 20:22:44')
```

**Comment by JoeCohen** [comment]
I asked Bing chat:
> In ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-darwin22], Rails 7.1.2, mysql  Ver 8.1.0 upon executing 
INSERT INTO `sequences` (`observation_id`, `user_id`, `locus`, `bases`, `archive`, `accession`, `notes`, `created_at`, `updated_at`) VALUES (516715, 4468, 'caron', 'atatat', '', '', 'ň', '2024-01-16 20:22:44', '2024-01-16 20:22:44')
I get this error
ActiveRecord::StatementInvalid: Mysql2::Error: Incorrect string value: '\xC5\x88' for column 'notes' at row 1

It answered:
>[The error Mysql2::Error: Incorrect string value happens when MySQL can’t insert the value you specified into the table because of incompatible encoding](https://sebhastian.com/mysql-incorrect-string-value/)[1](https://sebhastian.com/mysql-incorrect-string-value/). [In your case, the character ‘ň’ in the notes column is not supported by the default character set of MySQL, which is latin1](https://stackoverflow.com/questions/1168036/how-to-fix-incorrect-string-value-errors)[2](https://stackoverflow.com/questions/1168036/how-to-fix-incorrect-string-value-errors).
>
>[To fix this error, you need to change the character set and collation of your database, table, or column to utf8mb4, which can handle 4-byte UTF-8 characters](https://sebhastian.com/mysql-incorrect-string-value/)[1](https://sebhastian.com/mysql-incorrect-string-value/). 
You can do this by running the following SQL commands:
- To change the character set and collation of the whole database
```sql
ALTER DATABASE your_database_name CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;
```
- To change the character set and collation of a specific table
```sql
ALTER TABLE your_database_name.your_table CONVERT TO CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;
```
- To change the character set and collation of a specific column
```sql
ALTER TABLE your_database_name.your_table MODIFY your_column your_data_type CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;
```
>
>AI-generated code. Review and use carefully. [More info on FAQ](https://www.bing.com/new#faq).
>
>After changing the character set and collation, you should be able to insert the value ‘ň’ into the notes column without any error. I hope this helps you solve your problem. 😊

**Comment by nimmolo** [comment]
Sounds right, at least for that column. 
I'd research how changing a whole table, or the db, works with Rails, but i feel like UTF-8 is the one we'd want.

I also encountered something about this on the Unicorn email threads, when researching something unrelated. There was discussion about giving the server default encoding.

There may be other fields in the db that need to be UTF-8?

**Comment by JoeCohen** [comment]
Thanks Nimmo.
I've been looking into it.
All but 2 of our tables use utf8mb3
For reasons I don't understand both Articles and Sequences have charset:
"latin1". (These are both tables I created. Perhaps I specified latin1 when
I created them.)
There is documentation about how to write migrations which change
individual columns. But nothing about how to change entire tables (other
than changing every column) or the entire db via a migration.
I'll try to figure out if I can do it via a migration.
The (perhaps dangerous) alternative is to change it directly in the db.
